sklearn/linear_model/base.py

```diff
@@ -435,13 +507,12 @@ def fit(self, X, y, sample_weight=None):
         X, y = check_X_y(X, y, accept_sparse=['csr', 'csc', 'coo'],
                          y_numeric=True, multi_output=True)
 
-        if ((sample_weight is not None) and np.atleast_1d(
-                sample_weight).ndim > 1):
-            sample_weight = column_or_1d(sample_weight, warn=True)
+        if sample_weight is not None and np.atleast_1d(sample_weight).ndim > 1:
+            raise ValueError("Sample weights must be 1D array or scalar")
 
-        X, y, X_mean, y_mean, X_std = self._center_data(
-            X, y, self.fit_intercept, self.normalize, self.copy_X,
-            sample_weight=sample_weight)
+        X, y, X_mean, y_mean, X_norm = self._preprocess_data(
```

Comment:
X_mean -> X_offset
y_mean -> y_offset
X_norm -> X_scale

do this change everywhere as X_mean here may not contain the actual mean of X.